activesupport/lib/active_support/encrypted_file.rb

```diff
@@ -36,7 +46,7 @@ def initialize(content_path:, key_path:, env_key:, raise_if_missing_key:)
     end
 
     def key
-      read_env_key || read_key_file || handle_missing_key
+      @key ||= read_env_key || read_key_file || handle_missing_key
```

Comment:
This method can return a string, nil, or raise an error, so memoising it seems a bit strange.

I guess the point is to avoid reading the key file multiple times - if so, perhaps memoise `read_key_file` instead?

```patch
diff --git a/activesupport/lib/active_support/encrypted_file.rb b/activesupport/lib/active_support/encrypted_file.rb
index ab8d2a5f0d..34cfae36c7 100644
--- a/activesupport/lib/active_support/encrypted_file.rb
+++ b/activesupport/lib/active_support/encrypted_file.rb
@@ -46,7 +43,7 @@ def initialize(content_path:, key_path:, env_key:, raise_if_missing_key:)
     end

     def key
-      @key ||= read_env_key || read_key_file || handle_missing_key
+      read_env_key || read_key_file || handle_missing_key
     end

     def read
@@ -102,7 +99,8 @@ def read_env_key
       end

       def read_key_file
-        key_path.binread.strip if key_path.exist?
+        return @key_file_contents if defined?(@key_file_contents)
+        @key_file_contents = key_path.binread.strip if key_path.exist?
       end

       def handle_missing_key
```